lib/parser.mly

```diff
@@ -110,10 +110,66 @@ stmt_list:
 
 stmt:
   | expr SEMI                               { Expr $1      }
-  | LBRACE stmt_list RBRACE                 { Block $2 }
-  | IF LPAREN expr RPAREN stmt ELSE stmt    { If($3, $5, $7) }
-  | WHILE LPAREN expr RPAREN stmt           { While ($3, $5)  }
+  | logic_expr                              { Expr $1 }
```

Comment:
this line looks weird to me, all the statements (except control blocks) should end with SEMI, or it is not a statement.

Reply:
thats not true though

if (x < y) {

}

not if (x < y); {

}

do you see what I mean? only in logical expressions, i.e. booleans will this be the case where we use the Expr with no semi.

I noticed this in the ternary operator this is how java does it: 

booleanExpression ? expression1 : expression2

there shouldn't be a semi in any of those cases.

Reply:
Yes, I understand why you added this `logic_expr`, and what you want to achieve totally makes sense. 
But it doesn't change the definition of a statement, if a clause is a statement, it supposes to end with semi, or it is a if/loop statement. 

We just need to figure out another way to achieve this.
I will say `expr QUESTION stmt COLON stmt` should be changed to `expr QUESTION expr COLON expr`, because the second and third clauses will return, so they are expression.

Reply:
sorry, I dont follow. maybe you could explain to me on discord